katago_server/games/models.py

```diff
@@ -64,25 +66,37 @@ class Game(Model):
     has_resigned = BooleanField()
     score = DecimalField(max_digits=4, decimal_places=1)
 
+    created = DateTimeField(auto_now_add=True)
+
     sgf_file = FileField()
 
     class Meta:
         abstract = True
 
 
 class Match(Game):
+    gating = ForeignKey(Gating, on_delete=PROTECT)
+
     class Meta:
         verbose_name_plural = 'Matches'
 
 
 class SelfPlay(Game):
+    network = ForeignKey(Network, on_delete=PROTECT)
+
     unpacked_training_file = FileField()
     packed_training_file = FileField()
 
+    game_extra_params = JSONField()
+
+
+class ForkedSelfPlay(Game):
```

Comment:
As I mentioned in other comments - I think we can simplify and eliminate entirely the notion of a "forked" game so long as it is understood that "game" might start in a position other than the empty position, and might start at a turn number other than 0. If we want to distinguish forked games from other games via plots, that can be done by a simple metadata, exactly the same way one would distinguish games with one komi from games with a different komi, or games with one rules with games with a different rules.

Reply:
Agreed, and i did remove forked game in a later commit.